BitwardenShared/UI/Auth/StartRegistration/StartRegistrationViewTests.swift

```diff
@@ -0,0 +1,112 @@
+        try textfield.setInput("text")
+        XCTAssertEqual(processor.dispatchedActions.last, .emailTextChanged("text"))
+    }
+
+    /// Updating the text field dispatches the `.nameTextChanged()` action.
+    func test_nameField_updateValue() throws {
+        let textfield = try subject.inspect().find(viewWithId: Localizations.name).textField()
+        try textfield.setInput("user name")
+        XCTAssertEqual(processor.dispatchedActions.last, .nameTextChanged("user name"))
+    }
+
+    /// Tapping the continue button performs the `.StartRegistration` effect.
+    func test_continueButton_tap() throws {
+        let button = try subject.inspect().find(button: Localizations.continue)
+        try button.tap()
+
+        waitFor(!processor.effects.isEmpty)
+
+        XCTAssertEqual(processor.effects.last, .startRegistration)
+    }
+
+    /// Tapping the region button dispatches the `.regionPressed` action.
+    func test_regionButton_tap() throws {
+        let button = try subject.inspect().find(
+            button: "\(Localizations.creatingOn): \(subject.store.state.region.baseUrlDescription)"
+        )
+        try button.tap()
+        waitFor(processor.dispatchedActions.last != nil)
+        XCTAssertEqual(processor.dispatchedActions.last, .regionTapped)
+    }
+
+    /// Tapping the receive marketing toggle dispatches the `.toggleReceiveMarketing()` action.
+    func test_receiveMarketingToggle_tap() throws {
+        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
+            throw XCTSkip("Unable to run test in iOS 16, keep an eye on ViewInspector to see if it gets updated.")
+        }
+        let toggle = try subject.inspect().find(viewWithId: ViewIdentifier.StartRegistration.receiveMarketing).toggle()
+        try toggle.tap()
+        XCTAssertEqual(processor.dispatchedActions.last, .toggleReceiveMarketing(true))
+    }
+
+    // MARK: Snapshots
+
+    /// Tests the view renders correctly when the text fields are all empty.
+    func test_snapshot_empty() {
+        assertSnapshot(matching: subject, as: .defaultPortrait)
+    }
+
+    /// Tests the view renders correctly when the text fields are all populated.
+    func test_snapshot_textFields_populated() throws {
+        processor.state.emailText = "email@example.com"
+        processor.state.nameText = "user name"
+
+        assertSnapshot(matching: subject, as: .defaultPortrait)
+    }
+
+    /// Tests the view renders correctly when the toggles are on.
+    func test_snapshot_toggles_on() throws {
+        processor.state.isReceiveMarketingToggleOn = true
+
+        assertSnapshot(matching: subject, as: .defaultPortrait)
+    }
+
+    /// Tests the view renders correctly when the marketing toggle is hidden.
+    func test_snapshot_environment_selfhosted() throws {
+        processor.state.showReceiveMarketingToggle = false
+
+        assertSnapshot(matching: subject, as: .defaultPortrait)
+    }
```

Comment:
🎨 Update naming as the file is names `...selfHosted` and it's showing `bitwarden.eu` and this test should be named because of the marketing toggle.